experimental/framework/tinylicious-client/src/TinyliciousClient.ts

```diff
@@ -50,16 +50,20 @@ export class TinyliciousClient {
     public async createContainer(
         serviceContainerConfig: TinyliciousContainerConfig,
         containerSchema: ContainerSchema,
+        preAttach?: (container: FluidContainer) => Promise<void>,
     ): Promise<TinyliciousResources> {
         const runtimeFactory = new DOProviderContainerRuntimeFactory(
             containerSchema,
         );
-        const container = await this.getContainerCore(
+        const detachedContainer = await this.getContainerCore(
             serviceContainerConfig,
             runtimeFactory,
             true,
         );
-        return this.getFluidContainerAndServices(container);
+        const {fluidContainer, containerServices} = await this.getFluidResources(detachedContainer);
+        if (preAttach) { await preAttach(fluidContainer); }
```

Comment:
nit: use new line with if statements. Surprised the linter didn't yell at you.

```suggestion
        if (preAttach) {
            await preAttach(fluidContainer);
        }
```